From f285853cad73fdd433c3ea48b179e8c94efc28c8 Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Tue, 15 Jul 2014 13:18:00 +0100
Subject: [PATCH 2/6] UPnP: Make sure that if no extensions handle sub-classed
 containers that we fall back to creating the parent
 container instead.

---
 mythtv/libs/libmythupnp/upnpcds.cpp |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythupnp/upnpcds.cpp b/mythtv/libs/libmythupnp/upnpcds.cpp
index 053c7a7..084cbea 100644
--- a/mythtv/libs/libmythupnp/upnpcds.cpp
+++ b/mythtv/libs/libmythupnp/upnpcds.cpp
@@ -1379,7 +1379,7 @@ CDSObject* UPnpCDSExtension::CreateContainer(const QString& sId,
 {
     CDSObject* pContainer = NULL;
 
-    if (sClass == "object.container.person")
+    if (sClass.startsWith("object.container.person"))
     {
         pContainer = CDSObject::CreatePerson( sId, sTitle, sParentId );
     }
@@ -1387,11 +1387,11 @@ CDSObject* UPnpCDSExtension::CreateContainer(const QString& sId,
     {
         pContainer = CDSObject::CreatePlaylistContainer( sId, sTitle, sParentId );
     }
-    else if (sClass == "object.container.album")
+    else if (sClass.startsWith("object.container.album"))
     {
         pContainer = CDSObject::CreateAlbum( sId, sTitle, sParentId );
     }
-    else if (sClass == "object.container.genre")
+    else if (sClass.startsWith("object.container.genre"))
     {
         pContainer = CDSObject::CreateGenre( sId, sTitle, sParentId );
     }
-- 
1.7.10.2

